v2/context/context.go

```diff
@@ -50,3 +52,24 @@ func TopicFrom(ctx context.Context) string {
 	}
 	return ""
 }
+
+// Opaque key type used to store backoff parameters
+type retryKeyType struct{}
+
+var retryKey = retryKeyType{}
+
+// WithRetry returns back a new context with the retry backoff parameters.
+func WithRetry(ctx context.Context, backoff types.Backoff) context.Context {
+	return context.WithValue(ctx, retryKey, backoff)
+}
+
+// RetryFrom looks in the given context and returns the backoff parameters if found, otherwise nil
+func RetryFrom(ctx context.Context) *types.Backoff {
```

Comment:
I know slinky recommended moving this to http but there are settings in other protocols that have similar settings, I feel like it is ok to keep it in the general spot here.

Reply:
Ok i'm fine with that